packages/query/src/common/compliance-checker/compliance-checker.hooks.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { describe, expect, it, vi } from 'vitest';
+
+import { useBreakOnNonCompliantEntity } from './compliance-checker.hooks';
+import { useCheckAddressComplianceQueries } from './compliance-checker.query';
+
+// Mock the useCheckAddressComplianceQueries hook
+vi.mock('./compliance-checker.query', () => ({
+  useCheckAddressComplianceQueries: vi.fn(),
+}));
+
+// hook rendering mock similar to React Testing Library
+function renderHook<T>(hookFn: (...args: any[]) => T) {
+  const result = { current: undefined as T | undefined };
+
+  // Function to run the hook and update the result
+  function rerender(...args: any[]) {
+    result.current = hookFn(...args);
+    return result;
+  }
+
+  // Initial render
+  rerender();
+
+  return {
+    result,
+    rerender,
+  };
+}
+
+const mockQueryResult = {
+  isSuccess: true,
+  isError: false,
+  isPending: false,
+  isLoading: false,
+  status: 'success',
+  error: null,
+  fetchStatus: 'idle',
+  isLoadingError: false,
+  isRefetchError: false,
+  dataUpdatedAt: Date.now(),
+  errorUpdatedAt: 0,
+  failureCount: 0,
+  failureReason: null,
+  errorUpdateCount: 0,
+  isInitialLoading: false,
+  isPaused: false,
+  promise: Promise.resolve(),
+  isFetched: true,
+  isFetchedAfterMount: true,
+  isFetching: false,
+  isPlaceholderData: false,
+  isPreviousData: false,
+  isRefetching: false,
+  isStale: false,
+  refetch: vi.fn(),
+  remove: vi.fn(),
+};
+
+describe('useBreakOnNonCompliantEntity', () => {
+  it('should return isCompliant true when no addresses are on sanctions list', () => {
+    // Mock implementation for compliant addresses
+    vi.mocked(useCheckAddressComplianceQueries).mockReturnValue([
+      // TODO investigate this type error / if its worth adding tests
+      // @ts-expect-error - mockQueryResult is not a UseQueryResult
```

Comment:
I tried generating tests for this and they work but I was battling types and gave up 🫨 

Maybe some TS wiz can help here but I think it's OK to suppress in a test for now